### PR TITLE
feat: Engine Tiers

### PR DESCRIPTION
**Pull request by arcoraven**
Add Engine tiers selection when adding an Engine instance.

image

---

## PR-Codex overview
This PR updates the Engine UI components, adds `EngineTier` types, and refactors the `CreateEngineInstanceButton`.

### Detailed summary
- Added `EngineTier` type for engine instances
- Updated UI components in `EngineNoConnectedWallet`
- Refactored `CreateEngineInstanceButton` logic for tier selection
- Added modal for choosing Engine deployment tier with pricing and features
- Introduced `EngineTierCard` component for displaying tier information

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/components/engine/create-engine-instance.tsx [hunk 1]
```diff
@@ -39,24 +45,28 @@ export const CreateEngineInstanceButton = ({
...
       category: "engine",
       action: "click",
       label: "clicked-cloud-hosted",
+      tier: engineTier,
     });
 
     try {
       const res = await fetch(
-        `${THIRDWEB_API_HOST}/v1/engine/add-cloud-hosted`,
+        `${THIRDWEB_API_HOST}/v1/engine/add-cloud-hosted-XXX`,
```
Comment (jnsdls) [inline src/components/engine/create-engine-instance.tsx:60]: WIP or is this actually the endpoint?

File: src/components/engine/create-engine-instance.tsx [hunk 2]
```diff
@@ -144,75 +152,169 @@ export const CreateEngineInstanceButton = ({
...
-      await onAddPaymentMethod();
+  const trackEvent = useTrack();
+  const router = useRouter();
+
+  const onClick = async (tier: EngineTier) => {
+    if (tier === "ENTERPRISE") {
+      trackEvent({
+        category: "engine",
+        action: "update-relayer",
+        label: "error",
```
Comment (jnsdls) [inline src/components/engine/create-engine-instance.tsx:169]: this seems like the wrong data to track here?

**Review by jnsdls — COMMENTED** [review]
mostly just questions for now

File: src/components/engine/create-engine-instance.tsx (hunk 1 above)
Comment (arcoraven) [reply src/components/engine/create-engine-instance.tsx:60]: Whoops WIP to not actually hit this flow

Comment (farhanW3) [reply src/components/engine/create-engine-instance.tsx]: fixed this

File: src/components/engine/create-engine-instance.tsx [hunk 3]
```diff
@@ -105,24 +115,22 @@ export const CreateEngineInstanceButton = ({
...
-              action: "click",
-              label: "clicked-add-payment-method",
-            });
-
-            // Switch the Cloud-hosted Engine modal with the Add Payment modal.
-            disclosure.onClose();
-            paymentDisclosure.onOpen();
+
+            if (accountQuery.data?.status === AccountStatus.ValidPayment) {
+              await addCloudHostedEngine();
```
Comment (jnsdls) [inline src/components/engine/create-engine-instance.tsx:124]: when this gets called the `setEngineTier()` from above is not set yet meaning the `engineTier` inside of that function will potentially use an incorrect value

Comment (jnsdls) [reply src/components/engine/create-engine-instance.tsx:124]: for a hacky fix pass the tier here directly
```suggestion
              await addCloudHostedEngine(tier);
```

File: src/components/engine/create-engine-instance.tsx [hunk 4]
```diff
@@ -39,12 +45,14 @@ export const CreateEngineInstanceButton = ({
   const trackEvent = useTrack();
   const toast = useToast();
   const accountQuery = useAccount();
+  const [engineTier, setEngineTier] = useState<EngineTier>("STARTER");
 
-  const requestCloudHostedEngine = async () => {
+  const addCloudHostedEngine = async () => {
```
Comment (jnsdls) [inline src/components/engine/create-engine-instance.tsx:50]: make this optionally accept tier
```suggestion
  const addCloudHostedEngine = async (tier?: EngineTier) => {
```

File: src/components/engine/create-engine-instance.tsx [hunk 5]
```diff
@@ -39,12 +45,14 @@ export const CreateEngineInstanceButton = ({
...
     trackEvent({
       category: "engine",
       action: "click",
       label: "clicked-cloud-hosted",
+      tier: engineTier,
```
Comment (jnsdls) [inline src/components/engine/create-engine-instance.tsx:55]: and then everywhere below *prefer* the explicitly passed value over the state value

File: src/components/engine/create-engine-instance.tsx [hunk 6]
```diff
@@ -47,12 +47,12 @@ export const CreateEngineInstanceButton = ({
...
   const [engineTier, setEngineTier] = useState<EngineTier>("STARTER");
 
-  const addCloudHostedEngine = async () => {
+  const addCloudHostedEngine = async (tier?: EngineTier) => {
     trackEvent({
       category: "engine",
       action: "click",
       label: "clicked-cloud-hosted",
-      tier: engineTier,
+      tier: tier ?? engineTier,
```
Comment (farhanW3) [inline src/components/engine/create-engine-instance.tsx:55]: Updated this too to keep the tracking correct with what is deployed

File: src/components/engine/create-engine-instance.tsx (hunk 5 above)
Comment (farhanW3) [reply src/components/engine/create-engine-instance.tsx]: I made to pass `tier ?? engineTier` to be consistent with what is deployed

File: src/components/engine/create-engine-instance.tsx (hunk 2 above)
Comment (jnsdls) [reply src/components/engine/create-engine-instance.tsx:169]: @farhanW3 can you update this, too - pretty sure it shouldn't be `update-relayer` action

**Review by jnsdls — CHANGES_REQUESTED** [review]
just the tracking data then its good

**Comment by jnsdls** [comment]
todo UI changes from testjam (on my plate)

**Review by jnsdls — APPROVED** [review]

**Pull request merged**